### PR TITLE
[Routing] Keeping routes priorities after add a name prefix to the collection

### LINKED ISSUE #37089: Routing Priority
**Symfony version(s) affected**: 5.1

**Description**  
Routing Priority doesn't support for controller with only have one method

**How to reproduce**  

`@Route("/abc/{id}")`
in : `XClassController::__invoke()`

`@Route("/abc/static-value-here", priority=7)
in: `YClassController::__invoke()`

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 5.1
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #37089
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Routing/RouteCollection.php [hunk 1]
```diff
@@ -179,8 +179,8 @@ public function addNamePrefix(string $prefix)
 
         foreach ($this->routes as $name => $route) {
             $prefixedRoutes[$prefix.$name] = $route;
-            if (null !== $name = $route->getDefault('_canonical_route')) {
-                $route->setDefault('_canonical_route', $prefix.$name);
+            if (null !== $canonicalName = $route->getDefault('_canonical_route')) {
```
Comment (yceruto) [inline src/Symfony/Component/Routing/RouteCollection.php:182]: Name collision

**Review by nicolas-grekas — APPROVED** [review]
Cool thanks for having a look

**Comment by yceruto** [comment]
This is called here for instance:
https://github.com/symfony/symfony/blob/6d6d989dbe96ef95fb7e5da34d7c1001e161d4c9/src/Symfony/Component/Routing/Loader/YamlFileLoader.php#L217-L219

even if the name prefix is not defined, due to:
https://github.com/symfony/symfony/blob/6d6d989dbe96ef95fb7e5da34d7c1001e161d4c9/src/Symfony/Component/Routing/Loader/YamlFileLoader.php#L174

I wonder if we should compare with empty string instead and save some ticks.

**Review by xabbuh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
@yceruto that'd make sense yes

File: src/Symfony/Component/Routing/Loader/YamlFileLoader.php [hunk 2]
```diff
@@ -171,7 +171,7 @@ protected function parseImport(RouteCollection $collection, array $config, strin
         $schemes = isset($config['schemes']) ? $config['schemes'] : null;
         $methods = isset($config['methods']) ? $config['methods'] : null;
         $trailingSlashOnRoot = $config['trailing_slash_on_root'] ?? true;
-        $namePrefix = $config['name_prefix'] ?? '';
+        $namePrefix = $config['name_prefix'] ?? null;
```
Comment (yceruto) [inline src/Symfony/Component/Routing/Loader/YamlFileLoader.php:174]: consistent with Xml loader: https://github.com/symfony/symfony/blob/6d6d989dbe96ef95fb7e5da34d7c1001e161d4c9/src/Symfony/Component/Routing/Loader/XmlFileLoader.php#L164

File: src/Symfony/Component/Routing/Tests/RouteCollectionTest.php [hunk 3]
```diff
@@ -362,5 +362,19 @@ public function testAddWithPriority()
         ];
 
         $this->assertSame($expected, $collection2->all());
+
+        $collection3 = new RouteCollection();
```
Comment (Tobion) [inline src/Symfony/Component/Routing/Tests/RouteCollectionTest.php:366]: please create a separate test function. this test does not need to be in the same test case

**Review by Tobion — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Pull request merged**